### ISSUE TITLE
speed of training

### ISSUE THREAD

**Issue by StefanBloemheuvel** [Graph-Machine-Learning-Group/grin#9]
Hi,

First of all, nice work! I am trying to figure out more what you did, so I want to run the experiments myself. 
However, I am not that familiar with Pytorch.
I got it running, but my time per epoch is around 1:10 minutes. 
Is this comparable to yours? Because otherwise there is something wrong with my GPU (or something else). 
It would be great to know because if 300 epochs have to be performed (as is done in your code), it would take around 6 hours to run:

```
python -m scripts.run_imputation --config config/grin/air36.yaml --in-sample False
```

Thanks in advance!

**Comment by marshka** [comment]
Hi,

I appreciate your interest in our work! I confirm that your running time is comparable to ours. Being a recurrent GNN, the time for a forward pass scales linearly with the length of the sequence, and cannot be parallelized.

Hope it clarifies your issue.

— Ivan

**Comment by andreacini** [comment]
Yes, training time can be quite long, a couple of notes: 1) usually you can get very good results with a much much lower number of epochs (you can also play with the learning rate scheduler), keep an eye on the validation loss curve; 2) I would rather use the updated code that we released in our library (https://github.com/TorchSpatiotemporal/tsl), which should run faster.

**Comment by StefanBloemheuvel** [comment]
Thank you both very much for the quick response to clarify my question, and also the redirection to the updated code!
